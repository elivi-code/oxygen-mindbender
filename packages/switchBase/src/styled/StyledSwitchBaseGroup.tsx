import styled from 'styled-components';

import SwitchBaseWrapper from './SwitchBaseWrapper';

export interface Props {
  isHorizontal: boolean;
}

const StyledSwitchBaseGroup = styled.fieldset<Props>`
  display: flex;
  flex-direction: ${({ isHorizontal }) => (isHorizontal ? 'row' : 'column')};
  border: none;
  margin: 0;
  padding: 0;
  ${SwitchBaseWrapper} ~ ${SwitchBaseWrapper} {
    ${({ isHorizontal, theme }) =>
      isHorizontal
        ? `margin-left: ${theme.horizontalGroupMargin}`
        : `margin-top: ${theme.verticalGroupMargin}`}
  }
  align-items: start;
`;

export default StyledSwitchBaseGroup;
