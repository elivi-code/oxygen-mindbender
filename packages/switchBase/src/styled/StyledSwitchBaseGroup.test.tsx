import React from 'react';
import renderer from 'react-test-renderer';

import StyledSwitchBaseGroup, { Props } from './StyledSwitchBaseGroup';

describe('<StyledSwitchBaseGroup>', () => {
  let wrapper;

  function setup(props: Props = { isHorizontal: false }) {
    return renderer.create(<StyledSwitchBaseGroup {...props} />).toJSON();
  }

  it('should style correctly the switch base group if it should be displayed horizontally', () => {
    wrapper = setup({ isHorizontal: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should style correctly the switch base group if it should not be displayed horizontally', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
