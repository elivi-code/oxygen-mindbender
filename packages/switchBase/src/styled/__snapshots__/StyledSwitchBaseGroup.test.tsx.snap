// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledSwitchBaseGroup> should style correctly the switch base group if it should be displayed horizontally 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border: none;
  margin: 0;
  padding: 0;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<fieldset
  className="c0"
/>
`;

exports[`<StyledSwitchBaseGroup> should style correctly the switch base group if it should not be displayed horizontally 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: none;
  margin: 0;
  padding: 0;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<fieldset
  className="c0"
/>
`;
